addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/AlertFiltersMultipleOptionsPanel.java

```diff
@@ -46,12 +46,15 @@
     private static final String REMOVE_DIALOG_CHECKBOX_LABEL =
             Constant.messages.getString("alertFilters.dialog.remove.checkbox.label");
 
+    private ExtensionAlertFilters extension;
     private DialogAddAlertFilter addDialog = null;
     private DialogModifyAlertFilter modifyDialog = null;
     private Context uiSharedContext;
 
-    public AlertFiltersMultipleOptionsPanel(AlertFilterTableModel model) {
+    public AlertFiltersMultipleOptionsPanel(
+            ExtensionAlertFilters extension, AlertFilterTableModel model) {
```

Comment:
Are we worried about changing this signature?

Do we really need to pass the extension around for code that is in the extension?

Reply:
I'm not, and we need a reference to it somehow. Pretty its done like that in a load of other extensions ;)